catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/ConfigurationStore.java

```diff
@@ -83,4 +96,127 @@ public void setInMemory(boolean isInMemory) {
   public void setForceAutoCommit(boolean forceAutoCommit) {
     this.forceAutoCommit = forceAutoCommit;
   }
+
+  public void setAnyTextFieldWhitelist(List<String> fieldWhitelist) {
+    this.anyTextFieldWhitelist.clear();
+    if (fieldWhitelist != null) {
+      this.anyTextFieldWhitelist.addAll(fieldWhitelist);
+    }
+
+    filterAnyTextFieldCache();
+  }
+
+  public List<String> getAnyTextFieldWhitelist() {
+    return new ArrayList<>(anyTextFieldWhitelist);
+  }
+
+  public void addAnyTextWhitelistField(String whitelistField) {
+    if (whitelistField != null) {
+      this.anyTextFieldWhitelist.add(whitelistField);
+    }
+
+    filterAnyTextFieldCache();
+  }
+
+  public void removeAnyTextWhiteListField(String whitelistField) {
+    this.anyTextFieldWhitelist.remove(whitelistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void setAnyTextFieldBlacklist(List<String> fieldBlacklist) {
+    this.anyTextFieldBlacklist.clear();
+    if (fieldBlacklist != null) {
+      this.anyTextFieldBlacklist.addAll(fieldBlacklist);
+    }
+
+    filterAnyTextFieldCache();
+  }
+
+  public List<String> getAnyTextFieldBlacklist() {
+    return new ArrayList<>(anyTextFieldBlacklist);
+  }
+
+  public void addAnyTextBlacklistField(String blacklistField) {
+    this.anyTextFieldBlacklist.add(blacklistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void removeAnyTextBlacklistField(String blacklistField) {
+    this.anyTextFieldBlacklist.remove(blacklistField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void addAnyTextField(String anyTextField) {
+    anyTextFieldsCache.add(anyTextField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void removeAnyTextField(String anyTextField) {
+    anyTextFieldsCache.remove(anyTextField);
+
+    filterAnyTextFieldCache();
+  }
+
+  public void clearAnyTextFieldCache() {
+    anyTextFieldsCache.clear();
+    filteredAnyTextFieldsCache.clear();
+  }
+
+  public Set<String> getAnyTextFieldsCache() {
+    return anyTextFieldsCache;
+  }
+
+  public Set<String> getFilteredAnyTextFields() {
+    return filteredAnyTextFieldsCache;
+  }
+
+  private void filterAnyTextFieldCache() {
```

Comment:
:pencil2: Ideally this `ConfigurationStore` would only have configuration as its responsibility.  I would recommend extracting the anyText field cache related fields, methods, and logic to a new class.

Reply:
I would be good with this being fixed on the master branch.